scripts/stage1.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+#
+# This script is run at package build time to create the base recovery
+# environment image. This image is then modified by the postinst script when
+# the package is installed, and the recovery_sync script is used on the
+# destination system to keep it up to date with configuration changes.
+#
+# shellcheck disable=SC2001
+
+set -euxo pipefail
+
+function die() {
+	echo "$1" >&2
+	exit 1
+}
+
+function get_deps() {
+	(ldd "$1" 2>/dev/null || true) | while read -r line; do
+		if ! echo "$line" | grep "=>" &>/dev/null; then
+			continue
+		fi
+		dep=$(echo "$line" | sed 's/.*=> \(.*\) (.*/\1/')
+		dn=$(dirname "$dep")
+		mkdir -p "$2/$dn"
+		rsync -aL "$dep" "$2/$dep"
```

Comment:
Is this going to rsync from the build system or from the `workdir`?

Reply:
I've tweaked this so it's going to copy preferentially from the workdir. It's unfortunately not possible to force ldd to only search in the workdir, but I can make it always search the workdir paths first.